Automattic/wp-calypso
# Site Settings: Move General settings to a separate section

This PR moves the General settings section to a separate chunk. This is the last large settings section that was not separated from the main settings chunk yet. Once we land this, we can perform several more cleanups within the `site-settings` code that are recommended after the recent changes.

Currently in production we have one `settings` chunk that is over 600K:
![](https://cldup.com/SPeaKVfS5Y.png)

This PR splits out the general section, so the resulting chunks are now close to half that size each:
![](https://cldup.com/eGXYlpVYmd.png)
![](https://cldup.com/tUyGfpKOhR.png)

In more detail, the PR does the following:

* Moves general section to a separate chunk, `settings-general`.
* Cleans up everything unnecessary from the `settings` section.
* Preloads the `settings-general` section on hovering the General item of the settings navigation. With this change, all of the settings navigation items are now preloaded on hover.
* Preloads the `settings-general` section on hovering the Settings item of the sidebar navigation (previously, the `settings` section was preloaded).
* Suggests a WP logo placeholder while sites are being loaded (previously we had a blank screen)

To test:

* Checkout this branch.
* Go to http://calypso.localhost:3000/settings/ and verify it redirects to /settings/general and asks you to pick a site. Pick a site and verify it loads the general section correctly.
* Test the following paths with both a JP and WP.com site and verify there are no regressions with them:
  * http://calypso.localhost:3000/settings/general/$site
  * http://calypso.localhost:3000/settings/delete-site/$site
  * http://calypso.localhost:3000/settings/start-over/$site
  * http://calypso.localhost:3000/settings/theme-setup/$site
* Play with settings in the General section and verify they are stored and retrieved correctly.
* Go to Writing settings and refresh the page. Verify the `settings-general` chunk is preloaded correctly when hovering the General settings nav menu item for the first time.
* Go to Posts and refresh the page. Verify the `settings-general` chunk is preloaded correctly when hovering the Settings settings sidebar menu item for the first time.
* Verify the WP logo placeholder appears properly in the General tab if the sites are not loaded yet.
